### DISCUSSION TITLE
4-phonon scattering rates

### DISCUSSION THREAD

**Discussion by cumtra** [nakib/elphbolt#122] [General]
Dear Nakib,
Thank you so much for your effort on a wonderful software!
I'm tring to calculate the four-phonon thermal conductivity in the latest release version (elphbolt-v1.1). I want to know what files are needed to calculate the four phonons. Is it necessary to follow this?
![WW32TGVNT5)}TXS)56A14FR](https://github.com/nakib/elphbolt/assets/147600604/cb88d729-f922-4020-9d12-968c2c80c4da)

Thanks!

**Comment by nakib** [comment]
Thanks for the kind words, cumtra! Regarding including the 4-phonon scattering rates in elphbolt: yes, at the moment you need to first calculate these with the `FourPhonon` code. The excerpt above gives all the necessary information. But let me know if anything is unclear. Some time soon I will add an example that includes 4-phonon scattering.

Best,
Nakib

**Comment by cumtra** [comment]
Dear Nakib,
I follow the manual to add the ‘FourPhonon_BTE.w_4ph_T0.300E+03’ and 'ourPhonon_BTE.qpoint_full' files,  and add  parameters fourph_mesh_ref = 4 and fourph=true to input.xml.  But there is still the following problem.
![3LSCG9D)QQA 9(PT)YS4A7X](https://github.com/nakib/elphbolt/assets/147600604/bf9e6c8e-fed3-4c85-a1c4-04f01a59b8ab)

**Reply by nakib** [reply]
Did you prepend at the top of this file the number of irreducible q-points that you had in your FourPhonon calculation?

**Comment by cumtra** [comment]
I prepend at the top of this file the number of irreducible q-points under your suggestion，and the problem was solved. But there are new mistakes(Wrong q-mesh in external 4-ph calculation. Exiting).  I set the q-mesh as follows: qmesh = 10 10 10; fourph_mesh_ref = 4.

**Reply by nakib** [reply]
From the following snippet it looks like the total number of q-points that you prepended at the top of your `FourPhonon_BTE.qpoints_full` file does not match the number of q-points calculated by elphbolt by dividing the transport q-mesh with number fourph_mesh_ref. Can you share the different wave vector meshes you are using?

```
       if(this_image() == 1 .and. num%fourph) then
          open(1, file="FourPhonon_BTE.qpoints_full", status="old")

          read(1, *) coarse_numq_full
          if(coarse_numq_full /= product(coarse_qmesh)) then
             call exit_with_message('Wrong q-mesh in external 4-ph calculation. Exiting.')
          end if
       end if

       sync all
       call co_broadcast(coarse_numq_full, 1)
       sync all
```
